native/src/components/base/IconButton.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { ReactElement } from 'react'
+import { StyleProp, View, ViewStyle } from 'react-native'
+import styled from 'styled-components/native'
+
+import Pressable from './Pressable'
+
+const StyledPressable = styled(Pressable)<{ size: number }>`
+  height: ${props => props.size}px;
+  width: ${props => props.size}px;
+  align-items: center;
+  justify-content: center;
+  border-radius: ${props => props.size / 2}px;
+  background-color: ${props => props.theme.colors.backgroundColor};
+`
+
+type IconButtonProps = {
+  accessibilityLabel: string
+  icon: ReactElement
+  onPress: () => Promise<void> | void
+  size?: number
+  style?: StyleProp<ViewStyle>
+}
+
+const DEFAULT_BUTTON_SIZE = 40
+
+const IconButton = ({
+  accessibilityLabel,
+  icon,
+  onPress,
+  size = DEFAULT_BUTTON_SIZE,
```

Comment:
Not sure if we should use a prop for this as width/height/border-radius can easily be styled using styled components. My personal opinion is to not use a prop for this.